tests/unit/logic.test.js

```diff
@@ -0,0 +1,126 @@
+import { presentationToEEClaimGraph, proveh } from '../../src/utils/cd';
+import jsonld from 'jsonld';
+import { documentLoader } from '../cached-document-loader.js';
+import { randomAsHex } from '@polkadot/util-crypto';
+import { ANYCLAIM, MAYCLAIM, MAYCLAIM_DEF_1, MAYCLAIM_DEF_2 } from '../../src/rdf-defs';
+
+const COOLNESS = 'https://example.com/coolnessLevel';
+
+describe('logic', () => {
+  test('delegation chain', async () => {
+    const cred = {
+      '@context': 'https://www.w3.org/2018/credentials/v1',
+      type: 'VerifiableCredential',
+      issuer: 'did:example:c',
+      credentialSubject: {
+        id: 'did:example:d',
+        [COOLNESS]: 1,
+      },
+    };
+
+    const del1 = {
+      '@context': 'https://www.w3.org/2018/credentials/v1',
+      type: 'VerifiableCredential',
+      issuer: 'did:example:a',
+      credentialSubject: {
+        id: 'did:example:b',
+        [MAYCLAIM]: { '@id': ANYCLAIM },
+      },
+    };
+
+    const del2 = {
+      '@context': [
+        'https://www.w3.org/2018/credentials/v1',
+        'https://www.w3.org/2018/credentials/examples/v1',
+      ],
+      type: 'VerifiableCredential',
+      issuer: 'did:example:b',
+      credentialSubject: {
+        id: 'did:example:c',
+        [MAYCLAIM]: { '@id': ANYCLAIM },
+      },
+    };
+
+    let to_prove = [[
+      { Iri: 'did:example:d' },
+      { Iri: COOLNESS },
+      { Literal: { datatype: 'http://www.w3.org/2001/XMLSchema#integer', value: '1' } },
+      { DefaultGraph: true },
+    ]];
+
+    let assumed = [[
+      { Iri: 'did:example:a' },
+      { Iri: MAYCLAIM },
+      { Iri: ANYCLAIM },
+      { DefaultGraph: true },
+    ]];
+
+    let cg;
+
+    // happy path
+    cg = cat(await asCg([cred, del1, del2]), assumed);
+    proveh(cg, to_prove, MAYCLAIM_DEF_1);
+    proveh(cg, to_prove, MAYCLAIM_DEF_2);
+
+    // the assumption assigning a root authority is not included in the claimgraph
+    cg = await asCg([cred, del1, del2]);
+    assert_error(
+      () => proveh(cg, to_prove, MAYCLAIM_DEF_1),
+      { CantProve: "NovelName" },
```

Comment:
Normally an unprovable composite claim will result in the error: `{ CantProve: "ExhaustedSearchSpace" }` but not in this specific case.

`{ DefaultGraph: true }` is never mentioned in the input `cg`. Neither is it mentioned in the output clause of `MAYCLAIM_DEF_1`. Rify will (correctly) infer that it is impossible to prove a composite claim containing `{ DefaultGraph: true }` so it will skip reasoning entirely.

https://github.com/docknetwork/rify/blob/ac12e6ad1fc23ac232e3d256e38edf9ad0ee69f7/src/prove.rs#L191